flocker/control/_model.py

```diff
@@ -617,9 +628,15 @@ class Deployment(PClass):
     :ivar PSet nodes: A set containing ``Node`` instances
         describing the configuration of each cooperating node.
     :ivar Leases leases: A map of ``Lease`` instances by dataset id.
+    :ivar PersistentState persistent_state: The non-discoverable persistent
+        state of the cluster. (Note: XXX This should idealy be a sibling to the
+        configuration of the cluster, instead of child; but the required
+        refactoring doesn't seem worth it currently, and can be done later if
+        ever).
```

Comment:
:grey_question: Ah cool, this is making a lot more sense to me, thanks for the new comments, they are great :+1: 

A couple questions here:

- You mention that ideally this would be sibling in configuration, what about if you instead shove `nodes` and `leases` down a layer of the hierarchy (under some new `configured_state` field)? Curious if you think that is more ugly or less ugly than current architecture, and how much work that is. More just exploring the design space here than making a real recommendation.

- Would you mind  adding a note that `leases` is configuration to the ivar docstring? I should have remembered that, but noting it here I think makes the "configure state that needs to be persisted" vs "non-configured persistent state" distinction more clear.